roles/validate_inventory/tasks/ai.yml

```diff
@@ -28,3 +28,56 @@
     fail_msg: "{{ item }} is not within the machine network!"
   when: vip_dhcp_allocation == false
   loop: "{{ groups['masters'] + (groups['workers'] | default([])) }}"  # This should not include day2_workers as they can be RWNs
+
+- name: Validate extra VIPs for dualstack
+  when:
+    - extra_api_vip is defined
+    - extra_api_vip | length > 0
+    - extra_ingress_vip is defined
+    - extra_ingress_vip | length > 0
+  block:
+    - name: Assert that Openshift version is supported for dualstack VIPs
+      assert:
+        that:
+          - openshift_full_version is version('4.12', '>=')
+        fail_msg: "openshift_full_version  must be >= 4.12. to support dualstack VIPs"
+
+    - name: Check if extra api VIP is within the extra machine networks
+      ansible.builtin.debug:
```

Comment:
just use `set_fact` here, if people need to see it they can use the vebosity flag to see the results of the task. 

```yaml
- name:  Check if extra api VIP is within the extra machine networks
  vars:
     extra_api_vip: "{{ hostvars['assisted_installer']['extra_api_vip'] }}"
  ansible.builtin.set_fact:
     extra_api_vip_tests: "{{ extra_api_vip_tests + [extra_api_vip | ansible.utils.ipaddr(item.cidr)]"
  loop: "{{ hostvars['assisted_installer']['extra_machine_networks'] }}"
```

Reply:
done, I squashed all commits and push latest changes in one.